src/inverted_index.c

```diff
@@ -404,7 +420,7 @@ int IR_Read(void *ctx, RSIndexResult **e) {
     goto eof;
   }
   do {
-    if (BufferReader_AtEnd(&ir->br)) {
+    while (BufferReader_AtEnd(&ir->br)) {
```

Comment:
i don't want a function as this is the only place it happens. The reason this shifted from an if to while, is that we might have empty index blocks, so assuming we're at the end and the next block is valid is not true anymore.